File: Android/coverage-java/app/src/test/java/com/example/ricardonuma/coverage/MainActivityUnitTest.java
package com.example.ricardonuma.coverage;

import android.view.View;
import android.widget.Button;

import org.hamcrest.core.Is;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricTestRunner;
import static org.hamcrest.MatcherAssert.assertThat;

@RunWith(RobolectricTestRunner.class)
public class MainActivityUnitTest {

    @Test
    public void shouldHideButtonAfterClick() {
        MainActivity activity = Robolectric.setupActivity(MainActivity.class);

        Button button = activity.findViewById(R.id.hide);
        button.performClick();

        assertThat(button.getVisibility(), Is.is(View.GONE));
    }
}

File: Android/coveragekotlin/app/src/test/java/com/example/ricardonuma/coverage_kotlin/MainActivityUnitTest.kt
package com.example.ricardonuma.coverage_kotlin

import android.view.View
import android.widget.Button
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.Robolectric
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class MainActivityUnitTest {

    @Test
    fun shouldHideButtonAfterClick() {
        val activity = Robolectric.setupActivity(MainActivity::class.java)

        val button = activity.findViewById<Button>(R.id.hide)
        button.performClick()

        MatcherAssert.assertThat(button.visibility, Matchers.`is`(View.GONE))
    }
}